controller/sync.go

```diff
@@ -687,7 +687,7 @@ func (sc *syncContext) runTasks(tasks syncTasks, dryRun bool) runState {
 				if result == v1alpha1.ResultCodeSyncFailed {
 					runState = failed
 				}
-				if !dryRun || result == v1alpha1.ResultCodeSyncFailed {
+				if !dryRun || sc.syncOp.DryRun || result == v1alpha1.ResultCodeSyncFailed {
```

Comment:
This is necessary because during a dry-run sync, we were not setting any results for the resource result.

Reply:
We reproduced this bug during workshop at kubecon :)